perf/src/bin/perf_server.rs

```diff
@@ -50,7 +50,7 @@ async fn main() {
 
 async fn run(opt: Opt) -> Result<()> {
     let (key, cert) = match (&opt.key, &opt.cert) {
-        (&Some(ref key), &Some(ref cert)) => {
+        (Some(key), Some(cert)) => {
```

Comment:
This doesn't seem to be fixing a lint. If it was, I would report a bug, as the code was intended as written. Match "ergonomics" impair readability.

Reply:
```
error: dereferencing a tuple pattern where every element takes a reference
  --> perf/src/bin/perf_server.rs:53:10
   |
53 |         (&Some(ref key), &Some(ref cert)) => {
   |          ^^^^^^^^^^^^^^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_borrowed_reference
   = note: `-D clippy::needless-borrowed-reference` implied by `-D warnings`
help: try removing the `&` and `ref` parts
   |
53 -         (&Some(ref key), &Some(ref cert)) => {
53 +         (Some(key), &Some(ref cert)) => {
   |

error: dereferencing a tuple pattern where every element takes a reference
  --> perf/src/bin/perf_server.rs:53:26
   |
53 |         (&Some(ref key), &Some(ref cert)) => {
   |                          ^^^^^^^^^^^^^^^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_borrowed_reference
help: try removing the `&` and `ref` parts
   |
53 -         (&Some(ref key), &Some(ref cert)) => {
53 +         (&Some(ref key), Some(cert)) => {
   |

error: could not compile `perf` due to 2 previous errors
warning: build failed, waiting for other jobs to finish...
error: could not compile `perf` due to 2 previous errors
```

This is the error output of `cargo clippy --all-targets -- -D warnings`.

Reply:
Thanks, opened https://github.com/rust-lang/rust-clippy/issues/10245. For now please allowlist the lint with a reference to that issue.